### ISSUE TITLE
What hooks exist for me to customize the base generated workspace?

### ISSUE THREAD

**Issue by chiubaka** [chiubaka/nx-plugin#16]
As I investigate for #11, a big question that's coming up is: how much control do I have over options passed to the base workspace generator to control workspace layout, etc?

My research so far suggests that I don't have that much control, really :/. I just get to post-process what comes out, which is a little annoying.

One thing to explore is whether or not modifying the `nx.json` file in the output project changes things.

**Comment by chiubaka** [comment]
Reading through the generator code for some of the other presets is helping here.

[Here's the one for the `react` preset](https://github.com/nrwl/nx/blob/master/packages/react/src/generators/init/init.ts). A bit odd to me that it's named `init`, but that seems to be an internal convention for them.

Most interesting in this one is the use of the `updateWorkspaceConfiguration` utility, which I think may actually give me more control over the base workspace.

**Comment by chiubaka** [comment]
Need to learn more about what the workspace configuration even is and what I can change / mess with here. Can likely read through the typings to understand that.

**Comment by chiubaka** [comment]
Ok, so using the `updateWorkspaceConfiguration` utility didn't actually change any of the base workspace generation lol. I think because the base workspace generation happens before my preset even gets called anyway.

I got curious about how / why whatever generated my `nx-plugin` has a different directory structure than what seems to be normally generated. Upon investigation, I've discovered that [`create-nx-plugin`](https://github.com/nrwl/nx/blob/master/packages/create-nx-plugin/bin/create-nx-plugin.ts) is actually just calling `nx new` to generate a workspace and then is post processing it by literally deleting directories and moving things around lol!

So in actuality, that's not so different from what I was initially trying to do in my preset... I'm starting to think this is the right mental model, though. Workspace generation has already completed, likely using the `empty` preset. I'm left with that to do whatever I want on top.
